streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -212,6 +213,7 @@ public void configure(final Map<String, ?> configs) {
         rebalanceProtocol = assignorConfiguration.rebalanceProtocol();
         taskAssignorSupplier = assignorConfiguration::taskAssignor;
         assignmentListener = assignorConfiguration.assignmentListener();
+        uniqueField = usedSubscriptionMetadataVersion >= 8 ? new byte[1] : new byte[0];
```

Comment:
This could be moved out to the `assignorConfiguration` class to match the other fields but the code is fairly straightforward so I left it here for now

Reply:
Sounds good. I think it's also fine to just initialize it to `new byte[1]` regardless of whether the version is high enough to actually need it or not. It's just a single byte, and simpler is better

Reply:
Also, now that I think about it, the `usedSubscriptionMetadataVersion` will only ever be >= 8 at this point. It might be set to something lower at some point later on, but it has to be at least 8 when the assignor is just being created/configured

Reply:
The autocode generator throws an error if `array.length != 0` and the version is < 8. It didn't seem to be a huge problem but there were some tests that used version 1 so initializing to `new byte[1]` threw errors

Reply:
That seems like an indication that the subscription is being encoded incorrectly. If the version is below 8, it shouldn't be trying to encode the `uniqueField` in the first place -- you shouldn't have to mimic that yourself by forcing it to be empty (and even if it is I worry there might be some additional bytes for metadata that would get stored for that field, when that field should not exist at all for versions < 8)